src/Domain/Syntax/Sort.php

```diff
@@ -14,13 +14,19 @@ class Sort
 
     private string $field;
 
-    private string $order;
+    private string|array $order;
 
-    public function __construct(string $field, string $order = self::ASCENDING)
+    public function __construct(string $field, string|SortOrder $order = self::ASCENDING)
     {
+        
         $this->field = $field;
-        $this->order = $order;
-        Assert::inArray($order, [self::ASCENDING, self::DESCENDING]);
+        
+        if (is_string($order)){
+            $this->order = $order;
+            Assert::inArray($order, [self::ASCENDING, self::DESCENDING]);
+        } else {
+            $this->order = $order->asArray();
+        }
```

Comment:
It would be nicer to have `$this->order` is always typed as SortOrder

```
    private SortOrder $order;

    public function __construct(string $field, string|SortOrder $order = self::ASCENDING)
    {

        $this->field = $field;

        if (is_string($order)) {
            Assert::inArray($order, [self::ASCENDING, self::DESCENDING]);
            $this->order = SortOrder::default();
        } else {
            $this->order = $order;
        }
```

This also means the `build()` need a `toArray()` on the order.